src/Particle/VirtualParticleSet.cpp

```diff
@@ -20,9 +20,23 @@
 #include "Particle/DistanceTableData.h"
 #include "Particle/createDistanceTable.h"
 #include "QMCHamiltonians/NLPPJob.h"
+#include "ResourceCollection.h"
 
 namespace qmcplusplus
 {
+
+struct VPMultiWalkerMem : public Resource
+{
+  /// multi walker reference particle
+  Vector<int, OffloadPinnedAllocator<int>> mw_refPctls;
+
+  VPMultiWalkerMem() : Resource("VPMultiWalkerMem") {}
+
+  VPMultiWalkerMem(const VPMultiWalkerMem&) : VPMultiWalkerMem() {}
+
+  Resource* makeClone() const override { return new VPMultiWalkerMem(*this); }
```

Comment:
It might be an over engineering in this particular case. My lesson learnt from other makeClone in QMCPACK is that I should strictly invoke the copy constructor.